docs/proposals/dockerfile_component_detection.md

```diff
@@ -0,0 +1,58 @@
+# Enhance port detection for dockerfiles that are not in a component.
+
+## Related GitHub Issues
+* https://github.com/redhat-developer/alizer/issues/235
+
+## Background
+
+What is a component?
+
+Components are defined in the `languages-customization.yml` as a language that has specfic `configuration_files`.
+
+Example of a Go component:
+
+```yaml
+Go:
+  exclude_folders:
+    - "vendor"
+  configuration_files:
+    - "go.mod"
+  component: true
+```
+
+What happens when the repo being scanned has component(s) found but also contains a dockerfile one level outside of the repo?
```

Comment:
Yes when it is right under the repo root. I'm not certain but I think this could happen in sub-dirs if the directory wasn't a component and only contained a Dockerfile and the outer path (repo root) wasn't detected as a component...

Something like:

```
.repo-being-detected-by-alizer
└── docker/
    └── Dockerfile
```

Will need to check what happens in this case

Reply:
There are some common locations where the dockerfile can be placed:
Dockerfile, docker/Dockerfile, .docker/Dockerfile, build/Dockerfile
(and the same set for Containerfile)
We should cover all those variants of common locations.